### PR TITLE
[yang]: Add yang models for table BGP_ALLOWED_PREFIXES

### PR DESCRIPTION
**Pull request by StormLiangMS**
#### Why I did it
end2end test is blocked by Yang model for BGP allowed prefix list.

#### How I did it
Create new yang files for BGP allowed prefix list, and add UT.

#### How to verify it
Run UT for sonic-yang-models.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### A picture of a cute animal (not mandatory but encouraged)
Fix: #9866

### CODE REVIEW

**Comment by StormLiangMS** [comment]
@wen587 I verified this on master image, looks ok now, the reason is that the previous key pattern is wrong, I use "prefixs_p4", now it works. 

admin@str-s6000-acs-11:~$ ls
add.json  verify_yang.py
admin@str-s6000-acs-11:~$ cat add.json 
{
    "BGP_ALLOWED_PREFIXES": {
        "DEPLOYMENT_ID|0": {
            "prefixes_v4": [
                "10.20.0.0/16"
            ],
            "prefixes_v6": [
                "fc02:20::/64"
            ]
        }
    }
}
admin@str-s6000-acs-11:~$ cat verify_yang.py 
import subprocess
import sonic_yang
import json
def get_config_db_json():
    cmd = "show runningconfiguration all"
    result = subprocess.Popen(cmd, shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
    text, err = result.communicate()
    return json.loads(text)
configDbJson = get_config_db_json()
sy = sonic_yang.SonicYang("/usr/local/yang-models")
sy.loadYangModel()
sy.loadData(configDbJson )
sy.validate_data_tree()
admin@str-s6000-acs-11:~$ python3 verify_yang.py 
sonic_yang(6):Note: Below table(s) have no YANG models: CONSOLE_SWITCH, CRM, DHCP_SERVER, FEATURE, KDUMP, RESTAPI, SNMP, SNMP_COMMUNITY, TELEMETRY, XCVRD_LOG
admin@str-s6000-acs-11:~$

**Review by wen587 — APPROVED** [review]
LGTM. Please wait for other reviewer's approval.

File: src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang:50]: Does community have any pattern? 123:123?

File: src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang:57]: BGP_ALLOWED_PREFIXES_COM_LIST and BGP_ALLOWED_PREFIXES_LIST have the same leaf: default_action, prefixes_v4 and prefixes_v6.
Can you use grouping?

Comment (StormLiangMS) [reply src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang:57]: @ganglyu good suggestion, but there is an issue when current sonic_yang_ext.py to tackle with leaf-list + grouping case, it failed to parse the yang models. I filed an issue to track this one.
https://github.com/Azure/sonic-buildimage/issues/10116

File: src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang
Comment (StormLiangMS) [reply src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang:50]: @ganglyu  good catch, I found there are at least 2 different formats for community id, one is 4 bytes, the other is 2 bytes:2bytes, I took a reference from existing yang model files, all of them take string as the type of community. 
I think we should put a enforcement for the community id for all cases, filed an issue to track this, address this may beyond the scope of this PR.
https://github.com/Azure/sonic-buildimage/issues/10145

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**
